src/infra/core/src/repos/dataset_repository_local_fs.rs

```diff
@@ -904,12 +904,12 @@ impl DatasetStorageStrategy for DatasetMultiTenantStorageStrategy {
         &self,
         raw_alias: &DatasetAlias,
     ) -> Result<DatasetAlias, ResolveDatasetError> {
-        Ok(if let Some(account_name) = &raw_alias.account_name {
-            let (_, canonical_account_name) = self.resolve_account_dir(account_name).int_err()?;
-            DatasetAlias::new(Some(canonical_account_name), raw_alias.dataset_name.clone())
-        } else {
-            raw_alias.clone()
-        })
+        let account_name = self.effective_account_name(raw_alias);
+        let (_, canonical_account_name) = self.resolve_account_dir(account_name).int_err()?;
```

Comment:
Q: if we get the `account_name` right after `effective_account_name()` call, then why do we need to call `resolve_account_dir()`?

Reply:
I guessed it covers some corner cases with checking path/updating alias prefix, that's why I left the call.
I believe @zaychenko-sergei can say more about that function.

Reply:
I don't really understand your change here, plz explain what you are trying to solve.

Reply:
the idea of the change was that there account should be always present in dataset alias in the multi-tenant workspace.